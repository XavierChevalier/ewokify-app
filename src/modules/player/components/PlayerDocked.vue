<script setup lang="ts">
  import { PropType } from 'vue'
  import { Track } from '@/modules/track/models/Track'
  import DotSeparator from '@/components/DotSeparator.vue'
  import PlayerProgressBar from '@/modules/player/components/PlayerProgressBar.vue'
  import IconLike from '@/components/IconLike.vue'
  import Icon from '@/components/Icon.vue'
  import { mdiPlayOutline } from '@mdi/js'
  import TrackCover from '@/modules/track/components/TrackCover.vue'
  import { isPropertyValid } from '@/tools/component-properties/PropertyValidator'
  import PlayerRemoteListeningDevice from '@/modules/player/components/PlayerRemoteListeningDevice.vue'

  defineProps({
    track: {
      type: Object as PropType<Track>,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: isPropertyValid(Track),
    },
    remoteListeningDevice: {
      type: String,
      required: false,
      default: null,
    },
  })
</script>

<template>
  <div>
    <TrackCover :track="track" />
    <span>{{ track.title }}</span>
    <DotSeparator />
    <span v-for="band in track.bands" :key="band">{{ band }}</span>
    <IconLike class="text-black" />
    <Icon :path="mdiPlayOutline" />
    <PlayerRemoteListeningDevice
      v-if="remoteListeningDevice"
      :remote-device-name="remoteListeningDevice"
    />
    <PlayerProgressBar :progress="50" />
  </div>
</template>
