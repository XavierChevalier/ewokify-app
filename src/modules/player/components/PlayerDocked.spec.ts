import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import {
  Default,
  ListeningOnDesktop,
} from '@/modules/player/components/PlayerDocked.stories'
import { ComponentTestsGenerator } from '@/tests/tests-generators/ComponentTestsGenerator'
import PlayerDocked from '@/modules/player/components/PlayerDocked.vue'

describe('PlayerDocked', () => {
  generateStorybookSnapshotTests({ Default, ListeningOnDesktop })

  const componentTestsGenerator = new ComponentTestsGenerator(PlayerDocked)
  componentTestsGenerator.itShouldBeDefined()
  componentTestsGenerator
    .property<string>('remoteListeningDevice')
    .itShouldBeDefined()
    .itShouldNotBeRequired()
    .itShouldHaveADefaultValue()

  componentTestsGenerator.generateTests()
})
