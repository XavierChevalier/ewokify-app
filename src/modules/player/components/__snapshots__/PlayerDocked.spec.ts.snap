// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDocked [snapshot] should render the "Default" component correctly 1`] = `
<div>
  <img
    alt="The track cover of ANSI.SYS • MASTER BOOT RECORD"
    class="rounded w-16 h-16"
    src="https://picsum.photos/50"
  />
  <span>
    ANSI.SYS
  </span>
  <span
    class="px-2"
  >
    •
  </span>
  
  <span>
    MASTER BOOT RECORD
  </span>
  
  <span
    class="flex text-neutral text-black"
  >
    <svg
      class=""
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12.1,18.55L12,18.65L11.89,18.55C7.14,14.24 4,11.39 4,8.5C4,6.5 5.5,5 7.5,5C9.04,5 10.54,6 11.07,7.36H12.93C13.46,6 14.96,5 16.5,5C18.5,5 20,6.5 20,8.5C20,11.39 16.86,14.24 12.1,18.55M16.5,3C14.76,3 13.09,3.81 12,5.08C10.91,3.81 9.24,3 7.5,3C4.42,3 2,5.41 2,8.5C2,12.27 5.4,15.36 10.55,20.03L12,21.35L13.45,20.03C18.6,15.36 22,12.27 22,8.5C22,5.41 19.58,3 16.5,3Z"
        style="fill: currentColor;"
      />
    </svg>
    <!--v-if-->
  </span>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M8.5,8.64L13.77,12L8.5,15.36V8.64M6.5,5V19L17.5,12"
      style="fill: currentColor;"
    />
  </svg>
  <!--v-if-->
  <div
    class="h-1 bg-primary-light"
  >
    <div
      class="h-full rounded-full bg-primary"
      style="width: 50%;"
    />
  </div>
</div>
`;

exports[`PlayerDocked [snapshot] should render the "ListeningOnDesktop" component correctly 1`] = `
<div>
  <img
    alt="The track cover of ANSI.SYS • MASTER BOOT RECORD"
    class="rounded w-16 h-16"
    src="https://picsum.photos/50"
  />
  <span>
    ANSI.SYS
  </span>
  <span
    class="px-2"
  >
    •
  </span>
  
  <span>
    MASTER BOOT RECORD
  </span>
  
  <span
    class="flex text-neutral text-black"
  >
    <svg
      class=""
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12.1,18.55L12,18.65L11.89,18.55C7.14,14.24 4,11.39 4,8.5C4,6.5 5.5,5 7.5,5C9.04,5 10.54,6 11.07,7.36H12.93C13.46,6 14.96,5 16.5,5C18.5,5 20,6.5 20,8.5C20,11.39 16.86,14.24 12.1,18.55M16.5,3C14.76,3 13.09,3.81 12,5.08C10.91,3.81 9.24,3 7.5,3C4.42,3 2,5.41 2,8.5C2,12.27 5.4,15.36 10.55,20.03L12,21.35L13.45,20.03C18.6,15.36 22,12.27 22,8.5C22,5.41 19.58,3 16.5,3Z"
        style="fill: currentColor;"
      />
    </svg>
    <!--v-if-->
  </span>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M8.5,8.64L13.77,12L8.5,15.36V8.64M6.5,5V19L17.5,12"
      style="fill: currentColor;"
    />
  </svg>
  <div />
  <div
    class="h-1 bg-primary-light"
  >
    <div
      class="h-full rounded-full bg-primary"
      style="width: 50%;"
    />
  </div>
</div>
`;
